src/Middleware/HeaderPropagation/src/HeaderPropagationMessageHandler.cs

```diff
@@ -49,6 +64,12 @@ public HeaderPropagationMessageHandler(IOptions<HeaderPropagationOptions> option
         /// <returns>The task object representing the asynchronous operation.</returns>
         protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            var inputValues = _accessor?.HttpContext?.RequestServices.GetService<HeaderPropagationValues>() ??
```

Comment:
Nope, HttpContext is completely off limits here due to mutithreading.

Reply:
Will have to go bigger then....